File: frontend/src/lib/botIcons.ts
export interface BotIconOption { key: string; label: string; glyph: string; }

export const DEFAULT_BOT_ICON = "robot";
export const BOT_ICONS: BotIconOption[] = [
  { key: "robot", label: "Robot", glyph: "🤖" },
  { key: "briefcase", label: "Briefcase", glyph: "💼" },
  { key: "code", label: "Code", glyph: "💻" },
  { key: "search", label: "Search", glyph: "🔎" },
  { key: "test-tube", label: "Test tube", glyph: "🧪" },
  { key: "shield", label: "Shield", glyph: "🛡️" },
  { key: "sparkles", label: "Sparkles", glyph: "✨" },
  { key: "chat", label: "Chat", glyph: "💬" },
  { key: "brain", label: "Brain", glyph: "🧠" },
  { key: "rocket", label: "Rocket", glyph: "🚀" },
];

export function botIconFor(key: string | null | undefined): BotIconOption {
  return BOT_ICONS.find((icon) => icon.key === key) ?? BOT_ICONS[0];
}

File: backend/openbot/bot_icons.py
from __future__ import annotations

DEFAULT_BOT_ICON = "robot"
BOT_ICON_KEYS: tuple[str, ...] = (
    "robot",
    "briefcase",
    "code",
    "search",
    "test-tube",
    "shield",
    "sparkles",
    "chat",
    "brain",
    "rocket",
)


def validate_bot_icon(icon: str) -> str:
    if icon not in BOT_ICON_KEYS:
        raise ValueError(f"icon must be one of: {', '.join(BOT_ICON_KEYS)}")
    return icon
